### PR TITLE
Srsuddath search suggestions 21794

### PR DESCRIPTION
**Pull request by srsuddath**
## Description of change

This PR adds the search suggestions endpoint to vets-api. It requires a query, and uses an access key and name param provided by vets-api

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/21794

## Things to know about this PR

Settings.yml has been updated to include necessary parameters, swagger docs have also been updated to include the new endpoint

spec and configuration tests have been added, as well as manual testing of the endpoint via postman

### CODE REVIEW

**Comment by srsuddath** [comment]
@department-of-veterans-affairs/backend-review-group Hello all, I believe this PR to add the new search.gov suggestion endpoint to vets-api is ready for review!

**Comment by srsuddath** [comment]
This PR adds a new endpoint to vets-api, and as such cannot have its length reduced without cutting tests or other necessary features / functions

File: spec/lib/search_typeahead/service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          expect(response.status).to eq 200
+        end
+      end
+
+      it 'returns an array of suggestions' do
+        VCR.use_cassette('search_typeahead/success', VCR::MATCH_EVERYTHING) do
+          response = subject.suggestions
+          # rubocop:disable Layout/LineLength
+          # rubocop:disable Style/StringLiterals
+          expect(response.body).to eq "[\"ebenefits direct deposit\",\"ebenefits disability compensation\",\"ebenefits update contact information\",\"ebenefits your records\",\"ebenefits\"]"
```
Comment (ericboehs) [inline spec/lib/search_typeahead/service_spec.rb:29]:
```suggestion
          expect(JSON.parse(response.body)).to eq [
            'ebenefits direct deposit',
            'ebenefits disability compensation',
            'ebenefits update contact information',
            'ebenefits your records',
            'ebenefits'
          ]
```

**Review by ericboehs — APPROVED** [review]
Small suggestion.

File: spec/lib/search_typeahead/service_spec.rb (hunk 1 above)
Comment (srsuddath) [reply spec/lib/search_typeahead/service_spec.rb:29]: Perfect thats exactly what I needed.

**Pull request merged**